docs/atomic-swap-protocol-v1.md

```diff
@@ -0,0 +1,155 @@
+
+## The swap offer
+
+A swap offer is a HTLC with the following properties:
+
+- The amount of tokens is fixed. Implementation may support a group of multiple
+  amounts.
+- The recipient address is fixed.
+- Creating the contract locks the funds.
+- A hash of the preimage is fixed.
+- A timeout expressed as absolute time or block height is fixed.
+- At any point in time before the timeout, the swap offer can be claimed.
+- As soon as the timeout is reached, the funds can be returned to the creator.
+
+Common synonyms: _swap_, _contract_, _escrow_.
+
+### Preimage and hashing
+
+The preimage is 32 bytes of raw binary data. It is locally created by the long
+position and must be kept a secret. It is in the creator's own interest to use a
+sufficiently good source of entropy to generate it, but this is not required by
+the protocol.
+
+The hashing algorithm is fixed to SHA-256. The hash is 32 bytes of raw binary
+data with `hash = sha256(preimage)`.
+
+### Timeouts
+
+TODO
+
+### Swap offer actions
+
+Actions are mutations of the on-chain state triggered by transactions.
+
+- **Create:** By creating a swap offer, the funds are locked in the contract. A
+  timeout and a hash must be specified.
+- **Claim:** By revealing a preimage for the hash before the timeout is reached,
+  the contract transfers the tokens to the recipient. Requires a preimage of
+  exactly 32 bytes. Common synonyms: _release_, _execute_.
+- **Abort:** Withdraws the tokens from the contract to the creator. Requires
+  timeout to be reached.
+
+### Swap offer open state
+
+Each swap offer individually is either _open_, _claimed_ or _aborted_. The open
+state can only be changed by performing an action via a transaction.
+
+- **Open:** This is the initial open state.
+- **Claimed:** An _open_ swap becomes _claimed_ by performing the clain action.
+- **Aborted:** An _open_ swap becomes _aborted_ by performing the abort action.
+
+Common synonym: _settled = (claimed or aborted)_
+
+### Swap offer expiry state
+
+Each swap offer individually is either _non-expired_ or _expired_. The state
+automatically changes on-chain as the underlying time (wall time or block
+height) increases. A state can switch from _non-expired_ to _expired_ but not
+vice versa.
+
+- **Non-expired:** In the range `[creation, timeout)`, the swap offer is
+  _non-expired_. This is the default state for most offers. However, an offer
+  can be created in _expired_ state directly if the above interval is empty.
+- **Expired** In the range `[timeout, ∞)`, the swap offer is _expired_. Common
+  synonym: _abortable_.
+
+#### Note on open and expiry state
+
+Both states are independent of each other. A swap offer can be open and expired
+at the same time.
+
+## Risks
+
+The following risks for cross-chain atomic swaps have been identified. Some of
+them are fully avoided by this protocol but still documented for tranparancy.
+Others need to be mitigated by smart choice of parameters.
```

Comment:
suggest: Make it clear who's responsibility this is: "Others need to be mitigated by the user's choice of parameters."